internal/transformer-client/client.go

```diff
@@ -96,6 +109,67 @@ func NewClient(config *ClientConfig) Client {
 	}
 }
 
+func NewRetryableClient(config *RetryableClientConfig) RetryableClient {
```

Comment:
Not sure why we need this here. A `RetryableClient` just needs a http.Client & a bunch of configuration options as an input. Can't we create a retryable client in the webhook's setup instead, based on the http.Client that this package will produce?